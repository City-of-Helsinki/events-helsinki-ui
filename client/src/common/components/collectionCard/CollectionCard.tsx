import classNames from "classnames";
import { IconArrowRight } from "hds-react";
import React from "react";
import { Link } from "react-router-dom";

import { formatMessage } from "../../../common/translation/TranslationUtils";
import styles from "./collectionCard.module.scss";

export interface CollectionCardType {
  count: number;
  description: string;
  id: string;
  title: string;
}

export type CollectionCardSize = "sm" | "md" | "lg";

interface CollectionCardProps extends CollectionCardType {
  size: CollectionCardSize;
}

const CollectionCard: React.FC<CollectionCardProps> = ({
  count,
  description,
  id,
  size,
  title
}) => {
  return (
    <div
      className={classNames(
        styles.collectionCardWrapper,
        styles[`${size}Size`]
      )}
    >
      <div className={styles.imageWrapper}></div>
      <div className={styles.textWrapper}>
        <div className={styles.countWrapper}>
          <div className={styles.count}>
            {formatMessage("commons.collectionCard.count", { count })}
          </div>
        </div>

        <div className={styles.titleWrapper}>
          <div className={styles.title}>{title}</div>
          <div className={styles.description}>{description}</div>
        </div>

        <div className={styles.linkWrapper}>
          <Link to={`collection/${id}`}>
            <IconArrowRight />
          </Link>
        </div>
      </div>
    </div>
  );
};

export default CollectionCard;
