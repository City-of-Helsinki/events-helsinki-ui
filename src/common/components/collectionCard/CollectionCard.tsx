import classNames from 'classnames';
import { IconArrowRight } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory, useLocation } from 'react-router';
import { Link } from 'react-router-dom';

import { ROUTES } from '../../../domain/app/constants';
import useLocale from '../../../hooks/useLocale';
import IconButton from '../../components/iconButton/IconButton';
import TruncatedText from '../truncatedText/TruncatedText';
import styles from './collectionCard.module.scss';

export interface CollectionCardType {
  backgroundImage: string;
  count: number;
  description: string;
  showDescription?: boolean;
  slug: string;
  title: string;
}

export type CollectionCardSize = 'sm' | 'md' | 'lg';

export interface CollectionCardProps extends CollectionCardType {
  size: CollectionCardSize;
}

const CollectionCard: React.FC<CollectionCardProps> = ({
  backgroundImage,
  count,
  description,
  size,
  slug,
  showDescription = true,
  title,
}) => {
  const history = useHistory();
  const { search } = useLocation();
  const { t } = useTranslation();
  const locale = useLocale();
  const button = React.useRef<HTMLDivElement>(null);

  const collectionUrl = `/${locale}${ROUTES.COLLECTION.replace(
    ':slug',
    slug
  )}${search}`;

  const handleLinkClick = (ev: React.MouseEvent<HTMLAnchorElement>) => {
    const target = ev.target;
    if (button.current?.contains(target as Node)) {
      ev.preventDefault();
    }
  };

  const moveToCollectionPage = () => {
    history.push(collectionUrl);
  };

  return (
    <Link
      aria-label={t('commons.collectionCard.ariaLabelLink', {
        title,
      })}
      className={classNames(
        styles.collectionCardWrapper,
        styles[`${size}Size`]
      )}
      onClick={handleLinkClick}
      to={collectionUrl}
    >
      <div
        aria-hidden={true}
        className={styles.imageWrapper}
        style={{ backgroundImage: `url(${backgroundImage})` }}
      ></div>
      <div className={styles.textWrapper}>
        <div className={styles.countWrapper}>
          <div className={styles.count}>
            {t('commons.collectionCard.count', { count })}
          </div>
        </div>

        <div className={styles.titleWrapper}>
          <span className={styles.title}>{title}</span>
          {showDescription && !!description && (
            <TruncatedText
              as="div"
              className={styles.description}
              maxLength={120}
              text={description}
            />
          )}
        </div>

        <div className={styles.linkWrapper}>
          <div ref={button}>
            <IconButton
              ariaLabel={t('commons.collectionCard.ariaLabelLink', {
                title,
              })}
              aria-hidden={true}
              icon={<IconArrowRight />}
              onClick={moveToCollectionPage}
              size="default"
            />
          </div>
        </div>
      </div>
    </Link>
  );
};

export default CollectionCard;
