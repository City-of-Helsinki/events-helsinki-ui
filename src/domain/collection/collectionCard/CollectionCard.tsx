import classNames from 'classnames';
import { IconArrowRight } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory, useLocation } from 'react-router';
import { Link } from 'react-router-dom';

import IconButton from '../../../common/components/iconButton/IconButton';
import { CollectionFieldsFragment } from '../../../generated/graphql';
import useLocale from '../../../hooks/useLocale';
import { ROUTES } from '../../app/routes/constants';
import { getCollectionFields } from '../CollectionUtils';
import styles from './collectionCard.module.scss';

export type CollectionCardSize = 'sm' | 'md' | 'lg';

export interface Props {
  collection: CollectionFieldsFragment;
  size: CollectionCardSize;
}

const CollectionCard: React.FC<Props> = ({ collection, size }) => {
  const history = useHistory();
  const { search } = useLocation();
  const { t } = useTranslation();
  const locale = useLocale();
  const button = React.useRef<HTMLDivElement>(null);
  const {
    count,
    description,
    heroBackgroundImage: backgroundImage,
    slug,
    title,
  } = getCollectionFields(collection, locale);

  const collectionUrl = `/${locale}${ROUTES.COLLECTION.replace(
    ':slug',
    slug
  )}${search}`;

  const handleLinkClick = (ev: React.MouseEvent<HTMLAnchorElement>) => {
    const target = ev.target;
    if (button.current?.contains(target as Node)) {
      ev.preventDefault();
    }
  };

  const moveToCollectionPage = () => {
    history.push(collectionUrl);
  };

  const showDescription = size === 'lg';

  return (
    <Link
      aria-label={t('collection.collectionCard.ariaLabelLink', {
        title,
      })}
      className={classNames(
        styles.collectionCardWrapper,
        styles[`${size}Size`]
      )}
      onClick={handleLinkClick}
      to={collectionUrl}
    >
      <div
        aria-hidden={true}
        className={styles.imageWrapper}
        style={{ backgroundImage: `url(${backgroundImage})` }}
      ></div>
      <div className={styles.textWrapper}>
        <div className={styles.countWrapper}>
          <div className={styles.count}>
            {t('collection.collectionCard.count', { count })}
          </div>
        </div>

        <div className={styles.titleWrapper}>
          <span className={styles.title}>{title}</span>
          {showDescription && !!description && (
            <div className={styles.description}>{description}</div>
          )}
        </div>

        <div className={styles.linkWrapper}>
          <div ref={button}>
            <IconButton
              ariaLabel={t('collection.collectionCard.ariaLabelLink', {
                title,
              })}
              aria-hidden={true}
              icon={<IconArrowRight aria-hidden/>}
              onClick={moveToCollectionPage}
              size="default"
            />
          </div>
        </div>
      </div>
    </Link>
  );
};

export default CollectionCard;
