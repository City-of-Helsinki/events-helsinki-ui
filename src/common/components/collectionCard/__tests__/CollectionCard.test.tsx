import * as React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';

import CollectionCard from '../CollectionCard';

test('CollectionCard matches snapshot', () => {
  const component = renderer.create(
    <MemoryRouter>
      <CollectionCard
        backgroundImage={'imageurl.png'}
        count={120}
        description="Lorem ipsum"
        id="1"
        size="md"
        title="Test title"
      />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
