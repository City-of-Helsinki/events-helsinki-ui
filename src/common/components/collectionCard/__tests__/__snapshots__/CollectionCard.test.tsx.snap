// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionCard matches snapshot 1`] = `
<div
  className="collectionCardWrapper mdSize"
>
  <a
    aria-hidden={true}
    className="imageWrapper"
    href="/fi/collection/1"
    onClick={[Function]}
    tabIndex={-1}
  />
  <div
    className="textWrapper"
  >
    <div
      className="countWrapper"
    >
      <div
        className="count"
      >
        120 kpl
      </div>
    </div>
    <div
      className="titleWrapper"
    >
      <a
        aria-hidden={true}
        className="title"
        href="/fi/collection/1"
        onClick={[Function]}
        tabIndex={-1}
      >
        Test title
      </a>
      <div
        className="srOnly"
      >
        Test title
      </div>
      <div
        className="description"
      >
        Lorem ipsum
      </div>
    </div>
    <div
      className="linkWrapper"
    >
      <a
        href="/fi/collection/1"
        onClick={[Function]}
      >
        <svg
          className=""
          height="100%"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <polygon
              points="13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169"
            />
          </g>
        </svg>
      </a>
    </div>
  </div>
</div>
`;
